### PR TITLE
#7276 Fix mod copy functionality

### LINKED ISSUE #7276: Making a copy of a mod with changes clears changes in the original
## Describe the bug

- Reported by enterprise customer during office hours. Have not attempted to reproduce locally yet
- Making a copy of a mod with changes clears changes in the original. 

## To Reproduce

1. Activate a team mod
2. Make some changes in Page Editor
3. Click "make a copy" and make a copy

## Actual behavior

- Changes in the original mod are deleted
- You have 2 mods as expected

## Expected behavior

- Copy of mod has the changes applied. (Or at least you don't lose your work from the other mod)

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Fixes #7276 

## Checklist

- [ ] Add tests
- [ ] New files added to `src/tsconfig.strictNullChecks.json` (if possible)
- [x] Designate a primary reviewer - @grahamlangford

### CODE REVIEW

File: src/modDefinitions/modDefinitionsSlice.ts [hunk 1]
```diff
@@ -112,7 +112,12 @@ const loadModDefinitionsFromCache = createAsyncThunk<
 
   try {
     dispatch(modDefinitionsSlice.actions.startFetchingFromCache());
-    const modDefinitions = await modDefinitionRegistry.all();
+    const registryModDefinitions = await modDefinitionRegistry.all();
+    // Remove the top level registry item id to satisfy types properly
+    const modDefinitions: ModDefinition[] = registryModDefinitions.map((x) => {
+      const { id, ...rest } = x;
+      return rest;
+    });
```
Comment (BLoe) [inline src/modDefinitions/modDefinitionsSlice.ts:120]: These two similar changes are just cleanup to keep the objects strictly matching the types

**Comment by BLoe** [comment]
This isn't intended to merge until after #7299

File: src/pageEditor/slices/editorSlice.ts [hunk 2]
```diff
@@ -710,38 +710,6 @@ export const editorSlice = createSlice({
       state.visibleModalKey = ModalKey.CREATE_RECIPE;
       state.keepLocalCopyOnCreateRecipe = action.payload.keepLocalCopy;
     },
-    finishSaveAsNewRecipe(
```
Comment (twschiller) [inline src/pageEditor/slices/editorSlice.ts:713]: ❤️

File: src/modDefinitions/modDefinitionsSlice.ts [hunk 3]
```diff
@@ -133,7 +138,12 @@ export const syncRemoteModDefinitions = createAsyncThunk<
   try {
     dispatch(modDefinitionsSlice.actions.startFetchingFromRemote());
     await syncRemotePackages();
-    const modDefinitions = await modDefinitionRegistry.all();
+    const registryModDefinitions = await modDefinitionRegistry.all();
+    // Remove the top level registry item id to satisfy types properly
+    const modDefinitions: ModDefinition[] = registryModDefinitions.map((x) => {
+      const { id, ...rest } = x;
```
Comment (twschiller) [inline src/modDefinitions/modDefinitionsSlice.ts:144]: NIT: you could use lodash omit here

**Review by grahamlangford — APPROVED** [review]

**Review by twschiller — APPROVED** [review]
I'm OK merging this PR. There's a need to add test coverage for CreateRecipeModal in the future given how hairy the file is and how important save logic is. Right now it's sitting at 0% test coverage: https://app.codecov.io/gh/pixiebrix/pixiebrix-extension/blob/main/src%2FpageEditor%2Fsidebar%2Fmodals%2FCreateRecipeModal.tsx

**Comment by BLoe** [comment]
> There's a need to add test coverage for CreateRecipeModal

@twschiller FYI - There is some test coverage for both `useSaveMod` and `saveHelpers`, so not entirely zero coverage on the saving logic. We just don't have UI-level tests on the modal itself.

**Pull request merged**